utility/transaction.go

```diff
@@ -35,3 +38,25 @@ func (u *utilityModule) HandleTransaction(txProtoBytes []byte) error {
 	// Store the tx in the mempool
 	return u.mempool.AddTx(txProtoBytes)
 }
+
+// GetIndexedTransaction implements the exposed functionality of the shared utilityModule interface.
+func (u *utilityModule) GetIndexedTransaction(txProtoBytes []byte) (*coreTypes.IndexedTransaction, error) {
+	txHash := coreTypes.TxHash(txProtoBytes)
+	persistenceModule := u.GetBus().GetPersistenceModule()
```

Comment:
NIT: moving this closer to the point of usage (or even skipping the declaration altogether) would improve readability.